### ISSUE TITLE
Ability to spawn a server as a background process

### ISSUE THREAD

**Issue by sminnee** [silverstripe/silverstripe-serve#2]
In some cases, I'd like for some PHP code to be able to start up silverstripe/serve, have it run in the background, and then be shut down.

For example, it would be helpful if the testsession or behat-extension (or, presumably, undemanding) modules could work in concert with silverstripe/serve to start the server with an appropriate environment variable loaded into to provide access to the testsession database.

For such cases, it would be good if `SilverStripe\Serve\ServerLauncher` or similar could immediately return some kind of object. Perhaps symfony/process could be of use there?

**Comment by assertchris** [comment]
Check out https://github.com/assertchris/silverstripe-undemanding/blob/master/code/Listener.php

**Comment by sminnee** [comment]
Yeah, exactly :-) basically refactor that code into silverstripe/serve.

**Comment by sminnee** [comment]
Note that I would keep this separate from PHPUnit test cases.

I'd envision something like this:

``` php
$launcher = SilverStripe\Serve\ServerLauncher();
$server = $launcher->launchServer([ 'preferredPort' => 3000 ]);

echo "Server running on port " . $server->getPort() . " of " . $server->getHost() . "\n";  
echo file_get_contents($server->baseURL() . 'Security/login');
$server->stop();
```

Note that if we progressed with https://github.com/assertchris/silverstripe-serve/issues/2 we might also have an API such as this:

``` php
$launcher = SilverStripe\TestSession\TestSessionLauncher();
$testSession = $launcher->launchServer([ 'preferredPort' => 3000, 'fixtureFile' => 'something.yml' ]);

echo "Server running on port " . $testSession->getPort() . " of " . $testSession->getHost() . "\n";  
echo file_get_contents($testSession->baseURL() . 'Security/login');

$testSession->revertToFixture();

echo "Ready for another test!\n";

$testSession->stop();
```

**Comment by sminnee** [comment]
It would make sense for the result of `SilverStripe\Serve\ServerLauncher::launchServer()` and `SilverStripe\TestSession\TestSessionLauncher::launchServer()` to return an object of the same interface (or a common base class)

That way, something like undemanding could be passed such an object and easily switch between test-session-based and direct-database-based testing, without necessarily needing special handlers for it.

**Comment by sminnee** [comment]
This will be fixed by #7
